diff --git a/content_moderation_notifications.services.yml b/content_moderation_notifications.services.yml
--- a/content_moderation_notifications.services.yml
+++ b/content_moderation_notifications.services.yml
@@ -4,4 +4,4 @@
     arguments: ['@entity_type.manager', '@content_moderation.moderation_information']
   content_moderation_notifications.notification:
     class: Drupal\content_moderation_notifications\Notification
-    arguments: ['@current_user', '@entity_type.manager', '@plugin.manager.mail', '@module_handler', '@content_moderation_notifications.notification_information', '@renderer', '@config.factory', '@?token.entity_mapper']
+    arguments: ['@current_user', '@entity_type.manager', '@plugin.manager.mail', '@module_handler', '@content_moderation_notifications.notification_information', '@renderer', '@config.factory', '@token', '@?token.entity_mapper']
diff --git a/src/Notification.php b/src/Notification.php
--- a/src/Notification.php
+++ b/src/Notification.php
@@ -11,6 +11,7 @@
 use Drupal\Core\Mail\MailManagerInterface;
 use Drupal\Core\Render\RendererInterface;
 use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Utility\Token;
 use Drupal\token\TokenEntityMapperInterface;
 use Drupal\user\EntityOwnerInterface;
 use Drupal\user\RoleInterface;
@@ -28,6 +29,7 @@
     protected NotificationInformationInterface $notificationInformation,
     protected RendererInterface $renderer,
     protected ConfigFactoryInterface $configFactory,
+    protected Token $token,
     protected ?TokenEntityMapperInterface $tokenEntityMapper = NULL,
   ) {}
 
@@ -156,6 +158,9 @@
       );
 
+      // Replace any tokens.
+      $adhoc_emails = $this->token->replace($adhoc_emails, $data['params']['context']);
+
       // Split Adhoc emails on commas and newlines.
       $adhoc_emails = array_map('trim', explode(',', preg_replace("/((\r?\n)|(\r\n?))/", ',', $adhoc_emails)));
       foreach ($adhoc_emails as $email) {
         $data['to'][] = $email;
