diff --git a/src/Plugin/Transform/Type/RequestPath.php b/src/Plugin/Transform/Type/RequestPath.php
index 22ec447..16813e8 100644
--- a/src/Plugin/Transform/Type/RequestPath.php
+++ b/src/Plugin/Transform/Type/RequestPath.php
@@ -90,7 +90,7 @@ class RequestPath extends TransformationTypeBase {
       foreach ($this->transformBlocks->getVisibleBlocksPerRegion($cacheable_metadata_list)[$region] as $block) {
         $transformation[$block->id()] = BlockTransform::createFromBlock($block);
       }
-      $cacheMetadata = $cacheMetadata->merge($cacheable_metadata_list[$region]);
+      $cacheMetadata = $cacheMetadata->merge($cacheable_metadata_list[$region] ?? new CacheableMetadata());
     }
 
     $cacheMetadata->applyTo($transformation);
